Add baseline advisor for lr model

gale2307 opened the pull request.

aphedges reviewed (changes requested):
Sorry for taking so long to get back to you, but I needed to get some other stuff done first.

The good news is that I was able to run your bot and generate advice using it! No major changes are needed.

Here is some general feedback:

- Please switch from using `print()` to using the `logging` module in `baseline_models`. That will make debugging easier because we use `logging` in the bots, and mixing `print()` and `logging` can cause output to be out of order.
- Add a description of how to download the models and how to run the bots in the "Bots" section of the README.
	- You should include a link to <https://unisydneyedu-my.sharepoint.com/:f:/g/personal/nhad0493_uni_sydney_edu_au/EpvBJyx08X1HvnluND6tZAYBe2Fvoiz2GjVEM7Q_NpsAkg> in the README. You might also want to consider uploading the models to the CHIRON Google Drive so people can access it more easily.
	- This is a very minor concern, but it's confusing that the models are named using DDMMYYYY. If you use YYYYMMDD, then sorting the file names alphabetically would also sort them by date. You might want to consider renaming them.
- Does `KnnBot` run at all? If so, where should I get the model? If not, it should just be deleted.
- Run `make check` in the repository and fix the problems it finds. This appears to mostly be improving the docstrings.

gale2307 commented:
Hi Alex, thank you for the feedback!

I updated the docstrings and readme. We're not using the KNN bot anymore, so I deleted it. 
`make check` runs without errors as well.
I'll be working on changing print with logging in `baseline_models`, it should be done by sometime this week.

Let me know if there are any more changes to be made!

aphedges reviewed (approved):
I tested this out with `lr_20250114.zip`, and it looks good to me! I will go ahead and merge this PR, and I'll make the needed changes for it to build properly before we merge to `main`.
